src/components/AnimatedAppear/index.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { animated, useInView } from "react-spring";
+
+interface AnimatedAppearProps {
+    children: React.ReactNode;
+    className?: string;
+    onAppear?: () => void;
+}
+
+export function AnimatedAppear(props: AnimatedAppearProps) {
+    const [ref, styles] = useInView(
+        () => ({
+            from: {
+                opacity: 0,
+                y: 100,
+            },
+            to: {
+                opacity: 1,
+                y: 0,
+            },
+            onRest: () => {
+                props.onAppear?.();
+            },
+        }),
+        {
+            rootMargin: "100% 0% -10% 0%",
```

Comment:
How it works: https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver/rootMargin

Basically this defines at what point the element should be considered visible.